src/main/java/com/redhat/sast/api/service/AbstractBatchService.java

```diff
@@ -0,0 +1,222 @@
+                        i + 1,
+                        jobDtos.size(),
+                        packageNvr);
+
+                final Job createdJob = processor.createJob(jobDto);
+                jobId = createdJob.getId();
+                final List<Param> pipelineParams = processor.extractPipelineParams(createdJob);
+                final String llmSecretName = (createdJob.getJobSettings() != null)
+                        ? createdJob.getJobSettings().getSecretName()
+                        : "sast-ai-default-llm-creds";
+
+                batchProcessingService.associateJobToBatchInNewTransaction(jobId, batchId);
+
+                LOGGER.debug(
+                        "{} Batch {}: Starting pipeline for job {} (Package: {})",
+                        pipelineName,
+                        batchId,
+                        jobId,
+                        packageNvr);
+
+                platformService.startPipeline(jobId, processor.getPipelineName(), pipelineParams, llmSecretName);
+
+                boolean jobCompleted = batchProcessingService.waitForJobCompletion(jobId);
+
+                if (jobCompleted) {
+                    completedCount.incrementAndGet();
+                    LOGGER.debug(
+                            "{} Batch {}: Job {} completed successfully ({}/{} total)",
+                            pipelineName,
+                            batchId,
+                            jobId,
+                            completedCount.get(),
+                            jobDtos.size());
+                } else {
+                    failedCount.incrementAndGet();
+                    LOGGER.warn(
+                            "{} Batch {}: Job {} failed or timed out ({}/{} failed)",
+                            pipelineName,
+                            batchId,
+                            jobId,
+                            failedCount.get(),
+                            jobDtos.size());
+                }
+
+            } catch (Exception e) {
+                failedCount.incrementAndGet();
+                LOGGER.error(
+                        "{} Batch {}: Failed during processing of job {} for package {}",
+                        pipelineName,
+                        batchId,
+                        jobId,
+                        processor.getPackageNvr(jobDto),
+                        e);
+            } finally {
+                batchProcessingService.updateBatchProgressInNewTransaction(
+                        batchId, completedCount.get(), failedCount.get());
+            }
+        }
+
+        LOGGER.info(
+                "{} Batch {}: Sequential processing completed. Total: {}, Completed: {}, Failed: {}",
+                pipelineName,
+                batchId,
+                jobDtos.size(),
+                completedCount.get(),
+                failedCount.get());
+        batchProcessingService.finalizeBatch(batchId, jobDtos.size(), completedCount.get(), failedCount.get());
```

Comment:
I would not create a new method called "finalizeBatch", rather write the logic here. Because `updateBatchStatusInNewTransaction` and `updateBatchProgressInNewTransaction` are very similar. No need to have 2 methods. Let's just keep one method and `BatchStatus` creation can be done here.